pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -223,6 +223,16 @@ func (nc *NodeController) SyncCloudNodes() error {
 	// Create nodes which have been created in cloud, but not in kubernetes cluster.
 	for _, node := range matches.Items {
 		if _, ok := nodeMap[node.Name]; !ok {
+			glog.V(3).Infof("Querying addresses for new node: %s", node.Name)
+			nodeList := &api.NodeList{}
+			nodeList.Items = []api.Node{node}
+			_, err = nc.PopulateAddresses(nodeList)
+			if err != nil {
```

Comment:
Please add a comment saying that Nodes for which we fail populating address are skipped.